openldap/2.6/debian-11/rootfs/opt/bitnami/scripts/libopenldap.sh

```diff
@@ -0,0 +1,1045 @@
+            ldap_configure_tls
+        fi
+        # Initialize OpenLDAP with schemas/tree structure
+        if is_boolean_yes "$LDAP_ADD_SCHEMAS"; then
+            ldap_add_schemas
+        fi
+        if [[ -f "$LDAP_CUSTOM_SCHEMA_FILE" ]]; then
+            ldap_add_custom_schema
+        fi
+        if ! is_dir_empty "$LDAP_CUSTOM_SCHEMA_DIR"; then
+            ldap_add_custom_schemas
+        fi
+        # additional configuration
+        if [[ ! "$LDAP_PASSWORD_HASH" == "{SSHA}" ]]; then
+            ldap_configure_password_hash
+        fi
+        if is_boolean_yes "$LDAP_CONFIGURE_PPOLICY"; then
+            ldap_configure_ppolicy
+        fi
+        # enable accesslog overlay
+        if is_boolean_yes "$LDAP_ENABLE_ACCESSLOG"; then
+            ldap_enable_accesslog
+        fi
+        # enable syncprov overlay
+        if is_boolean_yes "$LDAP_ENABLE_SYNCPROV"; then
+            ldap_enable_syncprov
+        fi
+        # enable tls
+        if is_boolean_yes "$LDAP_ENABLE_TLS"; then
+            ldap_configure_tls
+            if is_boolean_yes "$LDAP_REQUIRE_TLS"; then
+                ldap_configure_tls_required
+            fi
+        fi
+        if ! is_dir_empty "$LDAP_CUSTOM_LDIF_DIR"; then
+            ldap_add_custom_ldifs
+        elif ! is_boolean_yes "$LDAP_SKIP_DEFAULT_TREE"; then
+            ldap_create_tree
+        else
+            info "Skipping default schemas/tree structure"
+        fi
+        info "OpenLDAP configuration and databases are now configured for service."
+        ldap_stop
+    else
+        info "Preserving existing config and data in ${LDAP_VOLUME_DIR}/{data,slapd.d}"
+    fi
+}
+
+########################
+# Run custom initialization scripts
+# Globals:
+#   LDAP_*
+# Arguments:
+#   None
+# Returns:
+#   None
+#########################
+ldap_custom_init_scripts() {
+    if [[ -f "${LDAP_VOLUME_DIR}/.user_scripts_initialized" ]]; then
+        info "Presence of file ${LDAP_VOLUME_DIR}/.user_scripts_initialized indicates that the user scripts have already been initialized."
+        return 0
+    fi
+    if is_dir_empty "${LDAP_ENTRYPOINT_INITDB_D_DIR}"; then
+        info "The user's custom files directory ${LDAP_ENTRYPOINT_INITDB_D_DIR} is missing or empty.";
+        return 0
+    fi
+    info "Loading user's custom files from ${LDAP_ENTRYPOINT_INITDB_D_DIR}";
+    if [[ -n $(find "${LDAP_ENTRYPOINT_INITDB_D_DIR}"/ -type f -regex ".*\.\(sh\)") ]]; then
+        for f in "${LDAP_ENTRYPOINT_INITDB_D_DIR}"/*; do
+            debug "Executing $f"
+            case "$f" in
+                *.sh)
+                    if [[ -x "$f" ]]; then
+                        if ! "$f"; then
+                            error "Failed executing $f"
+                            return 1
+                        fi
+                    else
+                        warn "Sourcing $f as it is not executable by the current user, any error may cause initialization to fail"
+                        . "$f"
```

Comment:
This is hacky, and if it's meant to be a part of the API now (even if unintentionally), the message should be clearer

Reply:
This is a hold-over from Bitnami's script; no changes from me (yet). A better approach is to exec things owned/executable by the current user, source files that are _not_ executable but _are_  owned by the current user, and ignore the rest.
```
    info "Loading user's custom files from ${LDAP_ENTRYPOINT_INITDB_D_DIR}";
    if [[ -n $(find "${LDAP_ENTRYPOINT_INITDB_D_DIR}"/ -type f -regex ".*\.\(sh\)") ]]; then
        for f in "${LDAP_ENTRYPOINT_INITDB_D_DIR}"/*; do
            case "$f" in
                *.sh)
                    if [[ -x "$f" ]]; then
			info "\texecuting $f"
			if [ ! "${SYMAS_DEBUG_SETUP:-}X" = "X" ]; then
			    bash -x "$f"
			else
			    bash "$f"
			fi
                        if [[ $? -nq 0 ]]; then
                            error "Failed executing $f"
                            return 1
                        fi
                    elif [[ -O "$f" ]]; then
			info "\tsourcing $f"
                        . "$f"
		    else
			warn "\tskipping $f because it is not owned by current user ($(id -u)/$(whoami)) and not executable"
                    fi
                    ;;
                *)
                    warn "\tskipping $f, supported formats are: .sh"
                    ;;
            esac
        done
```